### DISCUSSION TITLE
How to orchestrate queries in a single query editor?

### DISCUSSION THREAD

**Discussion by kuixiong** [getredash/redash#6108] [Q&A]
I am trying to run a presto query with redash, the query looks very much like
```
with temp1 as (
    select a_id, b_id, a_val, b_val from table1
)
select
    a_id, a_val/b_val
from (
    select a_id, a_val
    from table1 where a_id > 0
) a
join
(
    select b_id, b_val
    from table1 where b_id > 0
)b
on (a.a_id = b.b_id)
```

the table1 is very big, and with above query it will be scanned twice which is very inefficient. I want to avoid this kind of duplicated scanning by using a temp table, but I would also like to run the query with just one click. So it would be very convenient if redash query editor support orchestration. e.g 

```
create table temp1 as 
select a_id, a_val, b_id, b_val
from table1;

select
    a_id, a_val/b_val
from (
    select a_id, a_val
    from table1 where a_id > 0
) a
join
(
    select b_id, b_val
    from table1 where b_id > 0
)b
on (a.a_id = b.b_id);
```

After search, I realize redash actually supports querying result of another query, but that referenced query does not allow parameterization, that's really sad because I would like to integrate this query into a dashboard so we can view results by changing parameters.

Is there other way to achieve the result caching with redash? 🙏

**Comment by gaecoli** [comment]
Parameterization is supported here, maybe I didn't understand what you mean, I want to know more details. @kuixiong
